app/generate.js

```diff
@@ -110,13 +112,47 @@ function generateLangMenu(org, pages) {
       langs.push(lang);
     }
   }
-  if (langs.length <= 1) {
+
+  if (langs.length < 1) { //  defaultLang is not in langs array.
     return '';
   }
   const langMenuTemplate = template(fs.readFileSync(`./templates/${theme}/langMenu.tpl.html`, 'utf8'));
   return langMenuTemplate({ defaultLang, langs, webRoot });
 }
 
+function generatePageMenus(org, pages) {
+  const { theme } = org;
+  const retArray = [];
+  const langs = [];
+  const pageMenuTemplate = template(fs.readFileSync(`./templates/${theme}/pageMenu.tpl.html`, 'utf8'));
+
+  for (let i = 0; i < pages.length; i += 1) {
+    const { lang } = pages[i];
+    if (!langs.find(element => element.id === lang.id)) {
+      langs.push(lang);
+    }
+  }
+
+  for (let i = 0; i < langs.length; i += 1) {
+    const currentLang = langs[i];
+    const webRoot = org.defaultLang.id === currentLang.id ? '/' : `/${currentLang.link}/`;
+    let menuPages = [];
+
+    menuPages = pages.filter(page => page.lang.id === currentLang.id && page.menuStatus !== 'None');
```

Comment:
I would prefer `const menuPages = pages.filter(...);` instead of splitting this into two separate lines